3-0-spring-framework/3-2-1-account-rest-api/src/test/java/com/bobocode/AccountRestControllerTest.java

```diff
@@ -0,0 +1,164 @@
+package com.bobocode;
+
+import com.bobocode.config.RootConfig;
+import com.bobocode.config.WebConfig;
+import com.bobocode.dao.AccountDao;
+import com.bobocode.dao.impl.InMemoryAccountDao;
+import com.bobocode.model.Account;
+import com.bobocode.web.controller.AccountRestController;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.web.SpringJUnitWebConfig;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.lang.reflect.Constructor;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@SpringJUnitWebConfig(classes = {RootConfig.class, WebConfig.class})
+class AccountRestControllerTest {
+    @Autowired
+    private WebApplicationContext applicationContext;
+
+    @Autowired
+    private InMemoryAccountDao accountDao;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void setup() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(applicationContext).build();
+        accountDao.clear();
+    }
+
+    @Test
+    @Order(1)
+    @DisplayName("AccountRestController is marked as @RestController")
+    void accountRestControllerAnnotation() {
+        RestController restController = AccountRestController.class.getAnnotation(RestController.class);
+
+        assertNotNull(restController);
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("AccountRestController is annotated and mapped with @RequestMapping")
```

Comment:
@mathodcoast lets say `AccountRestController mapping is specified in @RequestMapping`